Documentation/design/launchphase.md

```diff
@@ -0,0 +1,90 @@
+# Launch
+
+## Goals
+
+1. Validate credentials for cloud.
+2. Create a functional cluster.
+3. Validate the cluster is ready to be used by an user.
+
+## Overview
+
+The launch phase creates a cluster based on the assets generated in the prepare phase. The launch performs operations,
+
+1. Pre-flight checks: Verify that credentials being passed to the installer are sufficient to bring up the cluster.
+2. Launch all the platform specific resources required for the cluster. For example in AWS this includes, DNS in Route53, VPCs, ELBs, IAM roles, Security groups etc.
+3. Bootstrap the cluster.
+4. Wait for the cluster to be ready for use by an user.
+
+## Detailed design
+
+### Idempotency
+
+Launch is **NOT** idempotent. Re-running the `installer launch` command should exit with a failure.
```

Comment:
Any reason why? As today the tectonic installer is idempotent right?

Reply:
today https://github.com/openshift/installer/blob/ae41b0a3fa6f3c074bb1e0fd726a9aec406c4301/installer/pkg/workflow/install.go#L129-L134 condition skips CNAME dance.

Reply:
It is not a requirement as we discussed previously. If the launch errors on re-run, it is acceptable.

Reply:
> As today the tectonic installer is idempotent right?

It is most definitely not idempotent. Running the installer a second time is one of the best ways to screw up your cluster.

Reply:
I did it couple times when I came back to my terminal from interruption, didn't remember seeing errors. Anyway whatever is easier is fine to me.